mypyc/test-data/irbuild-basic.test

```diff
@@ -196,24 +196,22 @@ def f(x: object, y: object) -> str:
 def f(x, y):
     x, y :: object
     r0 :: str
-    r1 :: int32
+    r1 :: int
     r2 :: bit
-    r3 :: bool
-    r4, r5 :: str
+    r3, r4 :: str
 L0:
     r0 = PyObject_Str(x)
-    r1 = PyObject_IsTrue(r0)
-    r2 = r1 >= 0 :: signed
-    r3 = truncate r1: int32 to builtins.bool
-    if r3 goto L1 else goto L2 :: bool
+    r1 = CPyObject_Size(r0)
```

Comment:
This is probably the bottleneck. This may be about as fast/slow as `PyObject_IsTrue`, so you aren't seeing an improvement. You'd need to specialize this operation for `str` values, and access the size of the string directly using C struct access. There are examples of how to do this for `list` and `tuple` values. Strings should be fairly similar (details may be different).